tune_sklearn/tune_search.py

```diff
@@ -380,27 +385,49 @@ def __init__(
             verbose=0,
             error_score="raise",
             return_train_score=False,
-            early_stopping=False,
             early_stopping_max_epochs=10,
     ):
         self.estimator = estimator
-        self.scheduler = scheduler
-        if self.scheduler is not None:
-            self.scheduler.metric = "average_test_score"
+        self.early_stopping = self._can_early_stop()
+        if self.early_stopping:
+            self.early_stopping_max_epochs = early_stopping_max_epochs
+            if isinstance(scheduler, str):
+                if scheduler in TuneBaseSearchCV.defined_schedulers:
+                    if scheduler == "PopulationBasedTraining":
+                        self.scheduler = PopulationBasedTraining(
+                            metric="average_test_score")
+                    elif scheduler == "AsyncHyperBandScheduler":
+                        self.scheduler = AsyncHyperBandScheduler(
+                            metric="average_test_score")
+                    elif scheduler == "HyperBandScheduler":
+                        self.scheduler = HyperBandScheduler(
+                            metric="average_test_score")
+                    elif scheduler == "HyperBandForBOHB":
+                        self.scheduler = HyperBandForBOHB(
+                            metric="average_test_score")
+                    elif scheduler == "MedianStoppingRule":
+                        self.scheduler = MedianStoppingRule(
+                            metric="average_test_score")
+                else:
+                    raise ValueError("{} is not a defined scheduler. "
+                                     "Check the list of available schedulers."
+                                     .format(scheduler))
+            else:
```

Comment:
Should check whether the scheduler is an instance of Ray Tune's `TrialScheduler` object, and error if the user passes in something besides a string or a `TrialScheduler`